cel-expr/cel-go
# Runtime cost calculation

Introduce runtime cost calculation into CEL.

The work is part of this [issue](https://github.com/kubernetes/kubernetes/issues/107573)

The current PR includes:

- Add runtime cost calculation into evaluation path
- Test to verify runtime cost calculation
- Test to verify runtime cost fell into estimation cost range
- Test to allow cost calculation for extension function

Todo in following PR:

- Add cancellation
- Add more test for large input
- Add test coverage for aligning runtime cost with estimate cost
